### PR TITLE
Add layout 4+2B right side 3QRT+1QRT

### PR DESCRIPTION
**Pull request by daniel-snowsurf**
Add a new layout 4+2 where the right side of the screen has two areas with 3 quarters and 1 quarter in height.
Changes apply to radio firmware and Simulator.

### CODE REVIEW

**Comment by daniel-snowsurf** [comment]
There is a problem in simulator (not tested on the radio). I can set and configure this layout, and it gets saved to model2.yml file. But when starting simulator again this layout (and the following layouts) doesn't show up (see last picture).
Model YML file: [model2.yml.txt](https://github.com/EdgeTX/edgetx/files/10908893/model2.yml.txt)

Image1
Image2
Image3
Image4
Image5 - Error missing views

**Comment by philmoz** [comment]
> There is a problem in simulator (not tested on the radio). I can set and configure this layout, and it gets saved to model2.yml file. But when starting simulator again this layout (and the following layouts) doesn't show up (see last picture). 

Was this using the standalone simulator or did you run the simulator from Companion?

If you start the simulator from Companion then changes made in the simulator are not saved on exit.

**Comment by pfeerick** [comment]
Good point. If you run Simulator via Companion, it is simulating just the `.etx` file you have created in Companion.

However, if you start the Firmware Simulator directly, any of the modes you run it in there (file, folder or sdcard) will write settings and thus persist across sessions. 

image

**Comment by daniel-snowsurf** [comment]
Just fixing it right now, it seems to be related to layout name length
`    #define LAYOUT_ID_LEN 11                   // Add 1 char more for Layout4P2B`
and tweek the checksizes
`    CHKSIZE(CustomScreenData, 851);           // Add 1 byte for CustomScreenData`
`    CHKSIZE(ModelData, 11133);                // Add 5 bytes for allCustomScreenData`
I'll confirm later.

**Comment by daniel-snowsurf** [comment]
I'm testing with Simulator directly, not using Companion.

**Comment by philmoz** [comment]
> Just fixing it right now, it seems to be related to layout name length
> ` #define LAYOUT_ID_LEN 11 // Add 1 char more for Layout4P2B`
> and tweek the checksizes
> ` CHKSIZE(CustomScreenData, 851); // Add 1 byte for CustomScreenData`
> ` CHKSIZE(ModelData, 11133); // Add 5 bytes for allCustomScreenData`
> I'll confirm later.

Might be better to set LAYOUT_ID_LEN to 12 to avoid potential data alignment issues in CustomScreenData.

**Comment by daniel-snowsurf** [comment]
OK, word aligned.

**Comment by pfeerick** [comment]
Just a FYI, any changes to `datastructs_private.h` will require `generate_yaml.h` to be re-run, as well as of the relevant `datastructs.h` tweaked - in this case it should only be the ones for the colorlcd radios, so the PCBHORUS and PCBNV14 groups.

**Comment by daniel-snowsurf** [comment]
I couldn't get the LayoutId "Layout4P2B" with 10 chars working, so I shortened the Id to "Layout4PB" with 9 chars. Now it works properly.

**Comment by elecpower** [comment]
@daniel-snowsurf  changing the radio data struct will also require a matching change in Companion or the field will be truncated on writing the yml files. For this PR you need to change this line
https://github.com/EdgeTX/edgetx/blob/a416548eccdb660b182bfc2f5c421c68948d44ad/companion/src/firmwares/customisation_data.h#L43

**Comment by daniel-snowsurf** [comment]
Thanks @elecpower, already found it, but changing LayoutId length is not working, The whole layout is not properly saved into the YAML file. The custom screen data is generated properly, but the YAML writer is not getting it right. Any clue?

**Comment by elecpower** [comment]
@daniel-snowsurf which yaml writer? Radio, Companion or both?

**Comment by philmoz** [comment]
AFAIK the LayoutID is only used when saving/loading from the yaml file - it is not shown to the user, so it can be anything we want it to be.

There is no reason for all the layout ID names to start with 'Layout' - shorten this to 'Lyt' and you have plenty of room for other stuff.

**Comment by pfeerick** [comment]
Good point... probably something for a future PR though, as it would need a custom reader added to convert `Layout` to `Lyt` on load on the radio, aide, as well as changes for Companion.

**Comment by philmoz** [comment]
> Good point... probably something for a future PR though, as it would need a custom reader added to convert `Layout` to `Lyt` on load on the radio, aide, as well as changes for Companion.

I wasn't suggesting changing the existing ones - just that any new layouts, like the one, can be more flexible with their naming.

**Comment by daniel-snowsurf** [comment]
Thanks @pfeerick for your work. The simulator works nicely with the T18 radio in Windows. With that I can update my models and widgets to EdgeTX. Waiting for the merge.
I've spend many hours trying to do it myself, but couldn't get the screenData size right.

**Comment by pfeerick** [comment]
Thanks for the confirmation. While I agree with Phil that the internal use layout names could be shorter, I think they should all be uniformly named so they are easily managed and no mistakes made later if they do get changed. i.e. it's the incremental changes that have a habit of shooting us in the foot later, three or four changes later.

**Comment by daniel-snowsurf** [comment]
Tested on Radioking TX18S, the new layout works fine.
